### PR TITLE
feat(atomic): added atomic-format-* support to atomic-result-number

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-900

Removed old formatting props from the v1 result number and added support for formatting components.

E.g.:
```html
<atomic-result-number field="sncost">
  <atomic-format-currency currency="CAD"></atomic-format-currency>
</atomic-result-number>
```

image

### CODE REVIEW

File: packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  Bindings,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {
+  defaultNumberFormatter,
+  NumberFormatter,
+} from '../../formats/format-common';
+
+/**
+ * The `atomic-result-number` component renders the value of a number result field.
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx:14]: Should probably mention the atomic-number-format component in the description

**Review by ThibodeauJF — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    const valueAsNumber = parseFloat(`${value}`);
+    return Number.isNaN(valueAsNumber) ? null : valueAsNumber;
+  }
+
+  private formatValue(value: number) {
+    try {
+      return this.formatter(value, this.bindings.i18n.languages);
+    } catch (error) {
+      this.error = new Error(
```
[the remark this answers is not in the document]
Comment (btaillon) [reply packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx:60]: Sure, I'll add it.

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/src/components.d.ts [hunk 3]
```diff
@@ -586,6 +586,12 @@ export namespace Components {
          */
         "minimumSignificantDigits"?: number;
     }
+    interface AtomicResultNumberV1 {
+        /**
+          * The result field which the component should use. This will look for the fields in the Result object first, and then in the Result.raw object. It is important to include the necessary field in the ResultList component.
```
Comment (dlutzCoveo) [inline packages/atomic/src/components.d.ts:591]:
```suggestion
   * The field that the component should use.
   * The component will try to find this field in the `Result.raw` object unless it finds it in the `Result` object first.
   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-result-list` component.
```

File: packages/atomic/src/components.d.ts [hunk 4]
```diff
@@ -1932,6 +1945,12 @@ declare namespace LocalJSX {
...
```
Comment (dlutzCoveo) [inline packages/atomic/src/components.d.ts:1950]:
```suggestion
   * The field that the component should use.
   * The component will try to find this field in the `Result.raw` object unless it finds it in the `Result` object first.
   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-result-list` component.
```

File: packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  @ResultContext() private result!: Result;
+
+  @Element() host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  /**
+   * The result field which the component should use.
+   * This will look for the fields in the Result object first, and then in the Result.raw object.
+   * It is important to include the necessary field in the ResultList component.
```
Comment (dlutzCoveo) [inline packages/atomic/src/components/result-template-components-v1/atomic-result-number/atomic-result-number.tsx:33]:
```suggestion
   * The field that the component should use.
   * The component will try to find this field in the `Result.raw` object unless it finds it in the `Result` object first.
   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-result-list` component.
```

**Review by dlutzCoveo — APPROVED** [review]

**Pull request merged**